pkgs/development/python-modules/openllm/default.nix

```diff
@@ -91,7 +91,7 @@ buildPythonPackage rec {
       # diffusers
       soundfile
       transformers
-    ] ++ transformers.optional-dependencies.agents;
+    ];
```

Comment:
Did you look into if we could add that field instead?

Reply:
Nope. I don't know how those attributes work or how to test them properly. I can only build them as is :)

It probably should provide something at  https://github.com/huggingface/transformers/blob/3cefac1d974db5e2825a0cb2b842883a628be7a0/setup.py#L412 . But that would be new functionality which a familiar maintainer should be able to test.

Reply:
> Nope. I don't know how those attributes work or how to test them properly.

optional-dependencies is just a list of python packages which can be optionally enabled. No magic and there is nothing really to test.



> a familiar maintainer should be able to test.

It's a python package. Likely you and me already know more than most person by just looking at this package. :)
---

We probably just need to add the following to transformers to make this work:

```nix
passthru = {
  optional-dependencies.agents = [ diffusers accelerate datasets torch sentencepiece opencv-python pillow ];
};
```
and maybe add a few new inputs.

Reply:
Sounds good. I'll give it a go. Converting to draft for now. If it works we might need to revert the other two PRs:
- https://github.com/NixOS/nixpkgs/pull/277788
- https://github.com/NixOS/nixpkgs/pull/277777

Reply:
Exposed `optional-dependencies.agents` to `python3Packages.transformers`.